test/functional/tests/cli/test_cli_standby.py

```diff
@@ -359,3 +367,95 @@ def test_activate_without_detach():
+        active_cls, standby_cls = CacheLineSize.LINE_4KiB, CacheLineSize.LINE_16KiB
+        cache_exp_obj_name = f"cas-cache-{cache_id}"
+
+        with TestRun.step("Start active cache instance."):
+            active_cache = casadm.start_cache(cache_dev=active_cache_dev, cache_id=cache_id,
+                                              cache_line_size=active_cls)
+
+        with TestRun.step("Create dump file with cache metadata"):
+            with TestRun.step("Get metadata size"):
+                dmesg_out = TestRun.executor.run_expect_success("dmesg").stdout
+                md_size = dmesg.get_metadata_size(dmesg_out)
+
+            with TestRun.step("Dump the metadata of the cache"):
+                dump_file_path = "/tmp/test_activate_corrupted.dump"
+                md_dump = File(dump_file_path)
+                md_dump.remove(force=True, ignore_errors=True)
+
+                dd_count = int(md_size / Size(1, Unit.MebiByte)) + 1
+                (
+                    Dd().input(active_cache_dev.path)
+                        .output(md_dump.full_path)
+                        .block_size(Size(1, Unit.MebiByte))
+                        .count(dd_count)
+                        .run()
+                )
+                md_dump.refresh_item()
+
+        with TestRun.step("Stop cache instance."):
+            active_cache.stop()
+
+        with TestRun.step("Start standby cache instance."):
+            standby_cache = casadm.standby_init(cache_dev=standby_cache_dev, cache_id=cache_id,
+                                                cache_line_size=int(
+                                                    standby_cls.value.value / Unit.KibiByte.value),
+                                                force=True)
+
+        with TestRun.step("Verify if the cache exported object appeared in the system"):
+            output = TestRun.executor.run_expect_success(
+                f"ls -la /dev/ | grep {cache_exp_obj_name}"
+            )
+            if output.stdout[0] != "b":
+                TestRun.fail("The cache exported object is not a block device")
+
+        with TestRun.step("Detach standby cache instance"):
+            standby_cache.standby_detach()
+
+        with TestRun.step(f"Copy changed metadata to the standby instance"):
+            Dd().input(md_dump.full_path).output(standby_cache_dev.path).run()
+            sync()
+
+        with TestRun.step("Try to activate cache instance"):
+            with pytest.raises(CmdException) as cmdExc:
+                output = standby_cache.standby_activate(standby_cache_dev)
+                if not check_stderr_msg(output, cache_line_size_mismatch):
+                    TestRun.LOGGER.error(
+                        f'Expected error message in format '
+                        f'"{cache_line_size_mismatch[0]}"'
+                        f'Got "{output.stderr}" instead.'
+                    )
+            assert "Failed to activate standby cache." in str(cmdExc.value)
```

Comment:
Shouldn't this assert be moved within `with` block?

Reply:
I'm asking because `cmdExc` was declared inside the `with` block but you are using it outside of it.

Reply:
According to pytest documentation this form is valid.